### PR TITLE
Switch to `packs`

### PR DESCRIPTION
**Pull request by iMacTia**
**WHY**

We recently updated `parse_packwerk`, `code_ownership` and other gems in the `rubyatscale` ecosystem.
They all moved to make `packs` a more central part of it, but this gem has not been updated yet, which causes issues like the following:

```ruby
VisualizePackwerk.team_graph!(teams)

# TypeError: Parameter 'package': Expected type Packs::Pack, got type ParsePackwerk::Package with value <ParsePackwerk::Package con...on" public_path="app/public/">
# Caller: ~/.rvm/gems/ruby-3.1.3/gems/visualize_packwerk-0.0.6/lib/visualize_packwerk/package_graph.rb:29
# Definition: ~/.rvm/gems/ruby-3.1.3/gems/code_ownership-1.29.3/lib/code_ownership.rb:162
```

**WHAT**

* Update the gem to be compatible with the latest version of the ecosystem's gems, including `code_ownership`.
* As a result of switching to `packs`, we now need to ignore the root package, but that was already in the plans based on comments I found in the code.
* Update the README to reflect the changes.
* ~Chore: Remove `Gemfile.lock` from VCS.~ Update `code_ownership` version in `Gemfile.lock`

**Testing**

Since this repo has no tests, I manually tested the following steps:
* `VisualizePackwerk.package_graph!` as demonstrated in the README
* `VisualizePackwerk.team_graph!` as demonstrated in the README
* The `modularity-graph` as explained in the `d3_graph_generator/README.md`

NOTE:
They were all successfully working against our codebase.
However, it's worth noting that we use an older version of `parse_packwerk` (0.14.0) since we're still using Packwerk 2.
It would be great to have someone else testing these changes against a Packwerk 3 codebase using the latest `parse_packwerk`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,10 @@
 /pkg/
 /spec/reports/
 /tmp/
+Gemfile.lock
```
Comment (alexevanczuk) [inline .gitignore:9]: > Chore: Remove Gemfile.lock from VCS. Although the Bundler team [now recommends on having it](https://bundler.io/guides/faq.html#using-gemfiles-inside-gems), that only makes sense with additional tooling in place (e.g. Dependabot). Without them, this is only making it harder to test against your dependencies and validate your gemspec properly.

Could you explain more context here? I think this is still useful to have in VCS to ensure all contributors are using the same set of dependencies, especially as it's recommended by bundler. In what ways are you finding it's harder to test against your dependencies and validate your gem spec?

File: README.md [hunk 2]
```diff
@@ -6,17 +6,17 @@ This gem contains rake tasks to help visualize relationships between packwerk pa
...
-selected_packs = ParsePackwerk.all.select{|p| ['Team 1', 'Team 2'].include?(CodeOwnership.for_package(p)&.name) }
+selected_packs = Packs.all
+selected_packs = Packs.all.select{ |p| ['packs/my_pack_1', 'packs/my_pack_2'].include?(p.name) }
+selected_packs = Packs.all.select{ |p| ['Team 1', 'Team 2'].include?(CodeOwnership.for_package(p)&.name) }
 VisualizePackwerk.package_graph!(selected_packs)
 ```
 
 # Building a team graph for specific teams
-```
+```ruby
```
Comment (alexevanczuk) [inline README.md:16]: Nice changes!

File: visualize_packwerk.gemspec [hunk 3]
```diff
@@ -24,12 +24,12 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version = '>= 2.6'
 
   spec.add_dependency 'sorbet-runtime'
+  spec.add_dependency 'packs'
   spec.add_dependency 'parse_packwerk'
   spec.add_dependency 'code_ownership'
   spec.add_dependency 'rake'
   spec.add_dependency 'ruby-graphviz'
 
-  spec.add_development_dependency 'bundler', '~> 2.2.16'
```
Comment (alexevanczuk) [inline visualize_packwerk.gemspec:32]: I think we'd still want this if we choose to continue to commit the `Gemfile.lock`.

**Review by alexevanczuk — COMMENTED** [review]
Thanks for this awesome contribution! Everything looks great – I'm just not sure about no longer checking in `Gemfile.lock`. Had some questions about that. If you wanted to continue to chat through it, I'd recommend we pull that change into a separate PR – that way this could be unblocked and we can go ahead and merge it!

Lastly, want to bump the minor version in the `gemspec`? That way CD will automatically publish a new release once this merges on `main` and CI passes.

**Comment by alexevanczuk** [comment]
Also looks like type check is failing! Let me know if I can help with that 😄

File: .gitignore (hunk 1 above)
Comment (iMacTia) [reply .gitignore:9]: Following your suggestion, I'll push the `Gemfile.lock` back into VCS and open a separate issue about this 👍

**Comment by iMacTia** [comment]
@alexevanczuk `Gemfile.lock` is back and sorbet looks happy when I run it locally.
Also, bumped the version in the `gemspec` to 0.1.0 🙌

**Review by alexevanczuk — APPROVED** [review]
Looks great, thank you!

**Pull request merged**